lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1074,19 +1107,22 @@ def deploy_vm(self):
             self.module.fail_json(msg='No folder matched the path: %(folder)s' % self.params)
         destfolder = f_obj
 
-        hostsystem = self.select_host()
-
-        if self.should_deploy_from_template():
+        if self.params['template']:
             # FIXME: need to search for this in the same way as guests to ensure accuracy
-            vm_obj = get_obj(self.content, [vim.VirtualMachine], self.params['template'])
-            if not vm_obj:
+            vm_obj = find_obj(self.content, [vim.VirtualMachine], self.params['template'])
+            if vm_obj is None:
                 self.module.fail_json(msg="Could not find a template named %(template)s" % self.params)
         else:
             vm_obj = None
 
+        if self.params['resource_pool']:
+            resource_pool = self.select_resource_pool_by_name(self.params['resource_pool'])
+
+            if resource_pool is None:
+                self.module.fail_json(msg='Unable to find resource pool "%(resource_pool)s"' % self.params)
+
         # set the destination datastore for VM & disks
         (datastore, datastore_name) = self.select_datastore(vm_obj)
-        resource_pool = self.select_resource_pool(hostsystem)
```

Comment:
@dagwieers The removal of this line causes the following error when cloning VMs on my company's vSphere 6.0 installation: `A specified parameter was not correct: ` (yes, that's the entire message... nothing follows the colon). Weirdly, this only happens on our Nehalem-based macOS cluster, but not our Haswell-based Linux/Windows cluster. Do you have idea why this might be happening?